packages/react-dom/package.json

```diff
@@ -47,21 +49,27 @@
     ".": "./index.js",
     "./client": "./client.js",
     "./server": {
+      "workerd": "./server.edge.js",
+      "edge-light": "./server.edge.js",
       "bun": "./server.bun.js",
       "deno": "./server.browser.js",
       "worker": "./server.browser.js",
       "browser": "./server.browser.js",
       "default": "./server.node.js"
     },
     "./server.browser": "./server.browser.js",
+    "./server.edge": "./server.edge.js",
     "./server.node": "./server.node.js",
     "./static": {
+      "workerd": "./static.edge.js",
+      "edge-light": "./static.edge.js",
       "deno": "./static.browser.js",
       "worker": "./static.browser.js",
```

Comment:
> We currently abuse the browser builds for Web streams derived environments. We already have a special build for Bun but we should also have one for [other "edge" runtimes](https://runtime-keys.proposal.wintercg.org/) so that we can maximally take advantage of the APIs that exist on each platform.

Wouldn't it make sense to also include the `worker` condition (that is defined here) when Next.js bundles for the `edge-light` runtime? The `worker` condition is IMHO closer to `edge-light` than the `browser` condition is to it. So it's a little bit of a weird situation that `browser` is automatically picked up but the `worker` is not.

For context - we've shipped the `worker` condition in Emotion in hope that it would get picked up by tools. There were no strong signals that other conditions would have to be needed and this seemed like a good "catch all" at the time.

What is different in the content of the files that you return here for the `edge-light` and `worker`?

Reply:
“worker” doesn’t have AsyncLocalStorage. The other two would include `require(“node:async_hooks”)` which would fail on worker builds.